src/Tizen.Messaging/Tizen.Messaging.Email/EmailSender.cs

```diff
@@ -40,35 +44,39 @@ public static async Task<EmailSendResult> SendAsync(EmailMessage email)
             email.FillHandle();
             email.Save();
 
-            Interop.Email.EmailSentCallback _emailSendingCallback = (IntPtr handle, int result, IntPtr userData) =>
+            int id = _callbackId++;
+            _sendCbMap[id] = (IntPtr handle, int result, IntPtr userData) =>
             {
-                task.SetResult((EmailSendResult)result);
+                task?.SetResult((EmailSendResult)result);
+                _sendCbMap.Remove((int)userData);
             };
 
-            ret = Interop.Email.SetCb(email._emailHandle, _emailSendingCallback, IntPtr.Zero);
+            ret = Interop.Email.SetCb(email._emailHandle, _sendCbMap[id], (IntPtr)id);
             if (ret != (int)EmailError.None)
             {
                 Log.Error(EmailErrorFactory.LogTag, "Failed to set email incoming callback, Error code: " + (EmailError)ret);
+                _sendCbMap.Remove(id);
                 throw EmailErrorFactory.GetException(ret);
             }
 
             ret = Interop.Email.SendEmail(email._emailHandle, saveToSentBox);
             if (ret != (int)EmailError.None)
             {
                 Log.Error(EmailErrorFactory.LogTag, "Failed to send email, Error code: " + (EmailError)ret);
+                _sendCbMap.Remove(id);
                 throw EmailErrorFactory.GetException(ret);
             }
 
-            var sendResult = await task.Task;
-
+            var sendResult = await task.Task.ConfigureAwait(false);
```

Comment:
Email API was not safe for multi-thread, so, you should return to main thread

```suggestion
            var sendResult = await task.Task.ConfigureAwait(true);
```